### PR TITLE
Add dvm to crab

### PR DESCRIPTION
**Pull request by AsceticBear**

### CODE REVIEW

**Comment by AsceticBear** [comment]
https://github.com/ethereum-lists/chains/pull/218

File: node/service/src/chain_spec/crab.rs [hunk 1]
```diff
@@ -33,6 +33,8 @@ use super::{
 };
 use crab_runtime::{constants::currency::COIN, *};
 use darwinia_primitives::{AccountId, Balance};
+// --- crates ---
```
Comment (AurevoirXavier) [inline node/service/src/chain_spec/crab.rs:36]: std

File: runtime/crab/src/lib.rs [hunk 2]
```diff
@@ -136,7 +138,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: sp_runtime::create_runtime_str!("Crab"),
 	impl_name: sp_runtime::create_runtime_str!("Darwinia Crab"),
 	authoring_version: 0,
-	spec_version: 42,
+	spec_version: 43,
```
Comment (AurevoirXavier) [inline runtime/crab/src/lib.rs:141]: Not a good idea to bump spec version within this PR

Comment (AsceticBear) [reply runtime/crab/src/lib.rs:141]: Yes,  the version change should be included in a separated pr. 

I am not sure which branch should be used when testing the DVM features in the pangolin network. 

1. Merge this pr into master firstly and testing use master branch
2. Directly use this pr branch to do test work, merge it after the test ends.

I prefer the 1. Test using master and add fix if something is broken.

Thus, The bump version change should be revert here. I will do it later after the locally test successfully.

Comment (AurevoirXavier) [reply runtime/crab/src/lib.rs:141]: 1 for final test (but not this PR)
2 for local test

**Review by HackFisher — COMMENTED** [review]
LGTM, please resolve conflicts before continue review and merge.

**Review by HackFisher — APPROVED** [review]
@AurevoirXavier If it looks good to you, please merge this(To your branch #641 )

File: node/service/src/service/crab.rs [hunk 3]
```diff
@@ -174,14 +210,25 @@ where
 	let rpc_setup = shared_voter_state.clone();
 	let babe_config = babe_link.config().clone();
 	let shared_epoch_changes = babe_link.epoch_changes().clone();
+	// <--- dvm ---
```
Comment (AurevoirXavier) [inline node/service/src/service/crab.rs:213]: I add these marks

Comment (AsceticBear) [reply node/service/src/service/crab.rs:213]: LGTM!

**Pull request merged**